samples/star-wars-api/Startup.fs

```diff
@@ -3,61 +3,51 @@ namespace FSharp.Data.GraphQL.Samples.StarWarsApi
-            .Configure(Action<IISServerOptions>(fun x -> x.AllowSynchronousIO <- true))
-            // Surprisingly minimal APIs use Microsoft.AspNetCore.Http.Json.JsonOptions
-            // Use if you want to return HTTP responses using minmal APIs IResult interface
-            .Configure<HttpClientJsonOptions>(
-                Action<HttpClientJsonOptions>(fun o ->
-                    Json.configureDefaultSerializerOptions Seq.empty o.SerializerOptions
-                )
-            )
-            // Use for pretty printing in logs
-            .Configure<HttpClientJsonOptions>(
-                Constants.Idented,
-                Action<HttpClientJsonOptions>(fun o ->
-                    Json.configureDefaultSerializerOptions Seq.empty o.SerializerOptions
-                    o.SerializerOptions.WriteIndented <- true
-                )
-            )
-            // Replace Newtonsoft.Json and use the same settings in Giraffe
-            .AddSingleton<Json.ISerializer, SystemTextJson.Serializer>(fun sp ->
-                let options = sp.GetService<IOptions<HttpClientJsonOptions>>()
-                SystemTextJson.Serializer(options.Value.SerializerOptions))
+            .AddGraphQLOptions<Root> (Schema.executor, rootFactory, "/ws")
         |> ignore
 
-    member _.Configure(app: IApplicationBuilder, env: IHostEnvironment) =
+    member _.Configure
+        (
+            app : IApplicationBuilder,
+            env : IHostEnvironment
+        ) =
         let errorHandler (ex : Exception) (log : ILogger) =
-            log.LogError(EventId(), ex, "An unhandled exception has occurred while executing the request.")
+            log.LogError (EventId (), ex, "An unhandled exception has occurred while executing the request.")
             clearResponse >=> setStatusCode 500
 
-        if env.IsDevelopment() then
-            app.UseGraphQLPlayground("/playground") |> ignore
-            app.UseGraphQLVoyager("/voyager") |> ignore
-            app.UseRouting() |> ignore
-            app.UseEndpoints(fun endpoints -> endpoints.MapBananaCakePop(PathString "/cakePop") |> ignore) |> ignore
+        if env.IsDevelopment () then
+            app.UseGraphQLPlayground ("/playground") |> ignore
+            app.UseGraphQLVoyager ("/voyager") |> ignore
+            app.UseRouting () |> ignore
+            app.UseEndpoints (fun endpoints -> endpoints.MapBananaCakePop (PathString "/cakePop") |> ignore)
+            |> ignore
 
         app
             .UseGiraffeErrorHandler(errorHandler)
             .UseWebSockets()
-            //.UseMiddleware<GraphQLWebSocketMiddleware<Root>>(Schema.executor, fun () -> { RequestId = Guid.NewGuid().ToString() })
-            .UseGiraffe HttpHandlers.webApp
+            .UseWebSocketsForGraphQL<Root>()
+            .UseGiraffe (
+                HttpHandlers.graphQL<Root>
+                >=> (setHttpHeader "Request-Type" "Classic")
```

Comment:
This won't work. Chaining another HttpHandler like you did here was my first shot to get the failing integration tests to pass again, but it didn't work, that's why I added that special function for customizing the response. I think it's because at this point the response stream has already been written to (by `HttpHandlers.graphQL<Root>`), so subsequent changes to the response headers won't be taken into consideration.

Reply:
Probably yes, what if we change the order? Set header first

Reply:
Attempted to fix